<?xml version="1.0"?>

<launch>
	<!-- <param name="use_sim_time" value="true"/> -->

<!--  <arg name="pc_bag_file" default="/media/biped/Cassie/bagfiles/2019-04-27/2019-04-27-14-38-10.bag" /> -->
  <!--<arg name="pc_bag_file" default="/home/biped/bagfiles/2019-06-10/belltower_test.bag" />   -->
  <?ignore <node pkg="rosbag" type="play" name="bagplayer" output="screen" args="--clock  -q $(arg pc_bag_file)"/>  ?>
  
  
  <!-- Visualize in rviz -->
  <!--<node pkg="rviz" type="rviz" name="rviz"  />-->

  <!-- <rosparam file="$(find SegmentationMapping)/data/nclt_04_29/label2color.yaml" command="load"/> -->

  <node pkg="SegmentationMapping" type="segmentation_node" name="segmentation_node" output="log" required="true">
    <!--<param name="neural_net_graph" type="str" value="$(find SegmentationMapping)/config/optimized_mobilenet_const_nclt_2019_06_11.pb"/>  -->
    <!-- <param name="neural_net_graph" type="str" value="$(find SegmentationMapping)/config/mobilenet_nclt/optimized_graph_mobilenet_trt32_2019-06-15.pb"/>  -->
    <!-- <param name="neural_net_graph" type="str" value="$(find SegmentationMapping)/config/extended_nclt/model_backup_4.hdf5"/>  -->
    <!-- <param name="neural_net_graph" type="str" value="$(find SegmentationMapping)/config/extended_nclt/extended_nclt_trained_weights.hdf5"/> -->
    <param name="neural_net_graph" type="str" value="$(find SegmentationMapping)/config/extended_nclt/gray_21_copy_rgb_jade_final.hdf5"/>
    <param name="network_image_input_tensor" type="str" value="network/input/Placeholder:0"/>
    <param name="network_distribution_output_tensor" type="str" value="network/output/ClassDistribution:0"/>
    <param name="network_label_output_tensor" type="str" value="network/output/ClassIndexPrediction:0"/>
    <!-- <param name="num_classes" type="int" value="14" /> -->
    <param name="num_classes" type="int" value="21" />
    <param name="skip_input_img_freq" type="int"    value="15" /> <!-- use every k frames -->>
    <param name="neural_net_input_width" type="int" value="640"/>
    <param name="neural_net_input_height" type="int" value="480"/>
    <param name="lidar_topic" type="str" value="/os_cloud_node/points_timestamp_corrected"/>
    <param name="enable_colored_labeling" type="bool" value="true" />
    <param name="is_input_gray" type="bool" value="true"/>
    <param name="rotate" type="bool" value="false"/>
    <param name="flip" type="bool" value="true"/>
    <param name="outdoor_threshold" type="float" value="0.3"/>

    <!-- input topic -->
    <remap from="~color_topic"        to="/other" />
    <remap from="~depth_topic"        to="/depth_aligned_to_other" />

  </node>

   <node pkg="SegmentationMapping" type="stereo_segmentation_node"  name="stereo_segmentation_node" output="log" required="true"  >
     <param name="color_topic"        type="string" value="/other" />
     <param name="depth_topic"        type="string" value="/depth_aligned_to_other" />
     <param name="depth_cam_topic"    type="string" value="/spot/camera/right/camera_info" />
     <param name="label_topic"        type="string" value="/segmentation_node/label_topic" />
     <param name="distribution_topic" type="string" value="/segmentation_node/distribution_topic" />
     <param name="skip_every_k_frame" type="int"    value="1" />
  </node> 


  <!-- Octomap, costmap generation -->
  <node pkg="SegmentationMapping" type="labeled_pc_map_node" name="labeled_pc_map_node" output="log" required="true" >
    <remap from="/labeled_pc_map_node/cost_map" to="/labeled_pc_map_node/cost_map2"/>
    <remap from="/labeled_pc_map_node/occupancy_grid" to="/labeled_pc_map_node/occupancy_grid2"/>

    <remap from="~cloud_in" to="/labeled_pointcloud" />
    <remap from="~cloud_out" to="/labeled_pointcloud_local_pc2" />

    <!-- <rosparam param="labels_to_ignore_in_map">[0, 8, 13]</rosparam>
    <rosparam param="target_labels_in_map">[2,3]</rosparam> -->

    <rosparam param="labels_to_ignore_in_map">[8, 10, 15]</rosparam>
    <rosparam param="target_labels_in_map">[0, 1, 14]</rosparam>


    <param name="distribution_enabled" type="bool" value="true" />
    <param name="static_frame"  type="string" value="odom" />
    <param name="body_frame"  type="string" value="camera_color_optical_frame" />
    <param name="painting_enabled" type="bool" value="true" />
    <param name="stacking_visualization_enabled" type="bool" value="false" />
    <param name="path_visualization_enabled" type="bool" value="true" />
    <param name="save_pcd_enabled" type="bool" value="false" />
    <param name="color_octomap_enabled" type="bool" value="false" />
    <param name="occupancy_grid_enabled" type="bool" value="false" />
    <param name="occupancy_grid_noise_percent" type="double" value="0.0" />
    <param name="cost_map_enabled" type="bool" value="false" />


    <!-- for semantic octomap generation -->
    <param name="octomap_enabled" type="bool" value="true"/>
    <param name="octomap_num_frames" type="int" value="3000" />
    <param name="octomap_max_dist" type="int" value="9" />
    <param name="octomap_resolution" type="double" value="0.20" />
    <param name="octomap_prob_hit" type="double" value="0.55" />
    <param name="octomap_prob_miss" type="double" value="0.3" />
    <param name="octomap_occupy_thresh" type="double" value="0.35" />
    <param name="bounding_box_size" type="double" value="7.0" />
    <param name="octomap_cropping_enabled" type="bool" value="true"/>
  </node>
  
</launch>

